### PR TITLE
Raspberry Pi OS 12 support

### PR DESCRIPTION
**Pull request by raulillo82**
## What does this PR change?

Add Raspbian OS 12 support for both arm64 and armhf architectures.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- Documentation issue needs to be created: https://github.com/uyuni-project/uyuni-docs/pull/2607 

- [X] **DONE**

## Test coverage
- Cucumber tests to be added later

- [ ] **DONE**

## Links

Implements #7853 

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: utils/spacewalk-common-channels.ini [hunk 1]
```diff
@@ -3675,6 +3675,253 @@ checksum = sha256
 base_channels = debian-12-pool-amd64-uyuni
 repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Debian12-Uyuni-Client-Tools/Debian_12/
 
+[raspbianos-12-pool-arm64-uyuni]
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:3678]: Unfortunately there's no RaspbianOS, name is Raspberry Pi OS: https://www.raspberrypi.com/software/operating-systems/ so we should use the official name here and at the other places (except when detecting the codename, as armhf still uses `raspbian` and arm64 uses `debian`).

File: spacewalk/certs-tools/rhn_bootstrap_strings.py [hunk 2]
```diff
@@ -441,7 +441,7 @@ def getRegistrationStackSh():
         elif [ -f /etc/redhat-release ]; then
             grep -v '^#' /etc/redhat-release | grep -q '\(Red Hat\)' && BASE="res"
             VERSION=`grep -v '^#' /etc/redhat-release | grep -Po '(?<=release )\d+'`
-	elif [ -f /etc/openEuler-release ]; then
+	    elif [ -f /etc/openEuler-release ]; then
```
Comment (juliogonzalez) [inline spacewalk/certs-tools/rhn_bootstrap_strings.py:444]: Looks to me this belongs to another PR?

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
Some small things, but in general the PR looks good.

File: utils/spacewalk-common-channels [hunk 3]
```diff
@@ -53,6 +53,8 @@ CHANNEL_ARCH = {
     'ppc64':        'channel-ppc64',
     'ppc64le':      'channel-ppc64le',
     'amd64-deb':    'channel-amd64-deb',
+    'arm64-deb':    'channel-arm64-deb',
+    'armhf-deb':    'channel-armhf-deb',
```
[the remark this answers is not in the document]
Comment (raulillo82) [reply utils/spacewalk-common-channels:57]: See https://github.com/uyuni-project/uyuni/issues/7857

File: spacewalk/certs-tools/rhn_bootstrap_strings.py (hunk 2 above)
Comment (raulillo82) [reply spacewalk/certs-tools/rhn_bootstrap_strings.py]: Restored. No idea why I was seeing a wrong tabulation... At any case, you're right, this is not the place to fix it.

File: utils/spacewalk-common-channels.ini (hunk 1 above)
Comment (raulillo82) [reply utils/spacewalk-common-channels.ini]: Changed everywhere (which were a lot of entries, I trusted search and replace + a manual inspection later on). Hope nothing's missing.

File: utils/spacewalk-common-channels.ini [hunk 4]
```diff
@@ -3675,6 +3675,253 @@ checksum = sha256
...
+repo_url = http://archive.raspberrypi.com/debian/dists/bookworm/main/binary-arm64/
+
+[raspberrypios-12-arm64-uyuni-client-devel]
+label    = raspberrypios-12-arm64-uyuni-client-devel
+name     = Uyuni Client Tools for Raspberry Pi OS 12 Bookworm ARM64 (Development)
+archs    = arm64-deb
+repo_type = deb
+checksum = sha256
+base_channels = raspberrypios-12-pool-arm64-uyuni
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/Debian12-Uyuni-Client-Tools/Debian_12/
```
Comment (deneb-alpha) [inline utils/spacewalk-common-channels.ini:3840]:
```suggestion
repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Debian12-Uyuni-Client-Tools/Debian_12/
```

**Comment by raulillo82** [comment]
All discussions solved as of now.

**Comment by raulillo82** [comment]
All discussions solved as of now.
I was told that the documentation was ready to be merged, but I asked to wait till this gets the approval and is merged before (or at the same time at least).
@juliogonzalez , could you have a look?

**Review by meaksh — APPROVED** [review]
Python files look good. It seems Salt preflight script is already taking care of Raspbian.

LGTM, thanks!

File: utils/spacewalk-common-channels.ini [hunk 5]
```diff
@@ -3675,6 +3675,253 @@ checksum = sha256
 base_channels = debian-12-pool-amd64-uyuni
 repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Debian12-Uyuni-Client-Tools/Debian_12/
 
+[raspberrypios-12-pool-arm64-uyuni]
+label    = raspberrypios-12-pool-arm64-uyuni
+checksum = sha256
+archs    = arm64-deb
+repo_type = deb
+name     = Raspberry Pi OS 12 (bookworm) main for arm64 for Uyuni
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:3683]:
```suggestion
name     = Raspberry Pi OS 12 (bookworm) main for ARM64 for Uyuni
```
Maybe?

I saw you used it in all other names.

File: utils/spacewalk-common-channels.ini [hunk 6]
```diff
@@ -3675,6 +3675,253 @@ checksum = sha256
...
+checksum = sha256
+base_channels = raspberrypios-12-pool-arm64-uyuni
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Debian12-Uyuni-Client-Tools/Debian_12/
+
+[raspberrypios-12-pool-armhf-uyuni]
+label    = raspberrypios-12-pool-armhf-uyuni
+checksum = sha256
+archs    = armhf-deb
+repo_type = deb
+name     = Raspberry Pi OS 12 (bookworm) pool for armhf for Uyuni
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:3856]:
```suggestion
name     = Raspberry Pi OS 12 (bookworm) pool for ARMHF for Uyuni
```

**Review by juliogonzalez — APPROVED** [review]
Just some minor nitpicking from this last review. Feel free to ignore if you want :-)
